src/modes/linear_world.cpp

```diff
@@ -262,6 +268,45 @@ void LinearWorld::updateGraphics(float dt)
 
 }   // updateGraphics
 
+// ----------------------------------------------------------------------------
+/** This calculate the time difference between the second kart in the
+ *  race and the first kart in the race (who must be a ghost)
+ */
+void LinearWorld::updateLiveDifference()
+{
+    // First check that the call requirements are verified
+    assert (race_manager->hasGhostKarts() && race_manager->getNumberOfKarts() >= 2);
+
+    AbstractKart* ghost_kart = getKart(0);
+
+    // Get the distance at which the second kart is
+    float second_kart_distance = getOverallDistance(1);
+
+    // Check when the ghost what at this position
+    float ghost_time;
+
+    // If there are two ghost karts, the view is set to kart 0,
+    // so switch roles in the comparison. Note that
+    // we can't simply multiply the time by -1, as they are assymetrical.
+    // When one kart don't increase its distance (rescue, etc),
+    // the difference increases linearly for one and jump for the other.
+    if (getKart(1)->isGhostKart())
```

Comment:
I'm not sure kart 1 is garanteed to exist

Reply:
Isn't it guaranteed to exist if we first guarantee that there are two karts, as is done here ?

Reply:
Ok I'd missed the >= 2 part above